cnd/src/db.rs

```diff
@@ -152,6 +159,53 @@ pub struct CreatedSwap<A, B> {
     pub role: Role,
 }
 
+// This only supports han-eth/halight-btc swap.
+/// Data learned during the communication protocols that enables a created swap
+/// to be finalized.
+#[derive(Debug, Clone, Copy)]
+pub struct FinalizedSwap {
+    /// The shared swap id, generated by Bob.
+    pub swap_id: SharedSwapId,
+    /// The secret hash, generated by Alice.
+    pub secret_hash: SecretHash,
+    /// The counterparties Ethereum identity.
+    pub ethereum_identity: identity::Ethereum,
+    /// the counterparties Lightning identity.
+    pub lightning_identity: identity::Lightning,
+}
+
+/// The trait that allows retrieval of a live swap.
+pub trait GetLiveSwap {
+    fn get_live_swap(&self, swap_id: LocalSwapId) -> anyhow::Result<LiveSwap>;
+}
+
+/// A 'live' swap is a swap that is currently live, i.e., the swap was 'created'
+/// by a POST on the REST API, 'finalized' during the communication protocols,
+/// and is now 'live'.
+#[derive(Debug)]
+pub struct LiveSwap {
+    pub alpha_ledger: Ethereum,
+    pub beta_ledger: lightning::Regtest,
+    pub alpha_asset: asset::Ether,
+    pub beta_asset: asset::Bitcoin,
+    pub alpha_ledger_refund_identity: identity::Ethereum,
+    pub alpha_ledger_redeem_identity: identity::Ethereum,
+    pub beta_ledger_refund_identity: identity::Lightning,
+    pub beta_ledger_redeem_identity: identity::Lightning,
+    pub alpha_expiry: Timestamp,
+    pub beta_expiry: Timestamp,
+    pub swap_id: LocalSwapId,
+    pub secret_hash: SecretHash,
+    pub secret: Option<Secret>,
```

Comment:
This is weird. Do we actually need all of this data in one place at some point? It is probably easier to name things if we don't create "god" structs that know everything but approach it more on a usecase basis.

For example the API never needs to know the secret hash as far as I know and neither does it need the identities.

Reply:
What separates a `LiveSwap` from a `FinalizedSwap`? Isn't it considered `live` once it is finalized? We would spawn the watchers once the party received and sent `finalized` (and only then a swap is `finalized` in my head). What am I missing?

Reply:
The idea was to have two types used for storing the data in the database (created and finalized) and then a type for retrieving the data (live).  This way _how_ the database stores the data is kept hidden within the database module.  The `LiveSwap` type added in this PR is just the data structure that was `FinalizedSwap` in the network code (created by lookups in the hash maps).  Since this is the functionality that we are trying to replace in #2545 it seems likely that we will need to expose exactly the same data from the db module as we do currently from the network module.  Does that make sense?  If this is not clear as a stand alone PR we (@da-kami and I) can just base our work on top of this and do a single large PR.